src/codeu/chat/server/Controller.java

```diff
@@ -62,42 +79,52 @@ public Message newMessage(
 
     Message message = null;
 
+
     if (foundUser != null && foundConversation != null && isIdFree(id)) {
 
       message = new Message(id, Uuid.NULL, Uuid.NULL, creationTime, author, body);
-      model.add(message);
-      LOG.info("Message added: %s", message.id);
 
-      // Find and update the previous "last" message so that it's "next" value
-      // will point to the new message.
+      if(database.write(message, foundConversation.id)){
+        model.add(message);
+        LOG.info("Message added: %s", message.id);
 
-      if (Uuid.equals(foundConversation.lastMessage, Uuid.NULL)) {
+        // Find and update the previous "last" message so that it's "next" value
+        // will point to the new message.
 
-        // The conversation has no messages in it, that's why the last message is NULL (the first
-        // message should be NULL too. Since there is no last message, then it is not possible
-        // to update the last message's "next" value.
+        if (Uuid.equals(foundConversation.lastMessage, Uuid.NULL)) {
 
-      } else {
-        final Message lastMessage = model.messageById().first(foundConversation.lastMessage);
-        lastMessage.next = message.id;
-      }
+          // The conversation has no messages in it, that's why the last message is NULL (the first
+          // message should be NULL too. Since there is no last message, then it is not possible
+          // to update the last message's "next" value.
+
+        } else {
+          final Message lastMessage = model.messageById().first(foundConversation.lastMessage);
+          lastMessage.next = message.id;
+        }
 
-      // If the first message points to NULL it means that the conversation was empty and that
-      // the first message should be set to the new message. Otherwise the message should
-      // not change.
+        // If the first message points to NULL it means that the conversation was empty and that
+        // the first message should be set to the new message. Otherwise the message should
+        // not change.
 
-      foundConversation.firstMessage =
-          Uuid.equals(foundConversation.firstMessage, Uuids.NULL)
-              ? message.id
-              : foundConversation.firstMessage;
+        foundConversation.firstMessage =
+                Uuid.equals(foundConversation.firstMessage, Uuid.NULL)
+                        ? message.id
+                        : foundConversation.firstMessage;
 
-      // Update the conversation to point to the new last message as it has changed.
+        // Update the conversation to point to the new last message as it has changed.
 
-      foundConversation.lastMessage = message.id;
+        foundConversation.lastMessage = message.id;
 
-      if (!foundConversation.users.contains(foundUser)) {
-        foundConversation.users.add(foundUser.id);
+        if (!foundConversation.users.contains(foundUser)) {
+          foundConversation.users.add(foundUser.id);
+        }
+      }else{
```

Comment:
Yeah, in the usual Java idioms, you shouldn't have to check boolean results like this; an exception should happen when there's a problem.  If you *do* need or want to do it like this, I'd handle the else part first and have it return; get that out of the way early.